internal/graph/reachableresources.go

```diff
@@ -203,69 +215,60 @@ func min(a, b int) int {
 	return a
 }
 
-func (crr *ConcurrentReachableResources) scheduleChunkedRedispatch(
-	t *TaskRunner,
+var queryLimit uint64 = 100
+
+func (crr *ConcurrentReachableResources) chunkedRedispatch(
+	ctx context.Context,
+	ci cursorInformation,
 	reader datastore.Reader,
 	subjectsFilter datastore.SubjectsFilter,
 	resourceType *core.RelationReference,
-	handler func(ctx context.Context, resources dispatchableResourcesSubjectMap) error,
-) {
-	t.Schedule(func(ctx context.Context) error {
-		toBeHandled := make([]resourcesSubjectMap, 0)
-		it, err := reader.ReverseQueryRelationships(
-			ctx,
-			subjectsFilter,
-			options.WithResRelation(&options.ResourceRelation{
-				Namespace: resourceType.Namespace,
-				Relation:  resourceType.Relation,
-			}),
-		)
-		if err != nil {
-			return err
-		}
-		defer it.Close()
-
-		rsm := newResourcesSubjectMap(resourceType)
-		chunkIndex := 0
-		for tpl := it.Next(); tpl != nil; tpl = it.Next() {
-			chunkSize := progressiveDispatchChunkSizes[min(chunkIndex, len(progressiveDispatchChunkSizes)-1)]
-			if it.Err() != nil {
-				return it.Err()
-			}
-
-			err := rsm.addRelationship(tpl)
+	handler func(ctx context.Context, ci cursorInformation, resources dispatchableResourcesSubjectMap) error,
+) error {
+	return withQueryInCursor(ci, "query-rels",
+		func(queryCursor options.Cursor, ci cursorInformation) (options.Cursor, error) {
+			it, err := reader.ReverseQueryRelationships(
+				ctx,
+				subjectsFilter,
+				options.WithResRelation(&options.ResourceRelation{
+					Namespace: resourceType.Namespace,
+					Relation:  resourceType.Relation,
+				}),
+				options.WithSortForReverse(options.ByResource),
+				options.WithAfterForReverse(queryCursor),
+				options.WithLimitForReverse(&queryLimit),
+			)
 			if err != nil {
-				return err
+				return nil, err
 			}
+			defer it.Close()
 
-			if rsm.len() == int(chunkSize) {
-				chunkIndex++
-				toBeHandled = append(toBeHandled, rsm)
-				rsm = newResourcesSubjectMap(resourceType)
-			}
-		}
-		it.Close()
+			rsm := newResourcesSubjectMap(resourceType)
+			var lastTpl *core.RelationTuple
+			for tpl := it.Next(); tpl != nil; tpl = it.Next() {
+				if it.Err() != nil {
```

Comment:
```go
if err := it.Err(); err != nil {
  return nil, err
}

Reply:
Why this vs just calling `it.Err` again?